File: app/src/main/java/com/raival/fileexplorer/common/view/BottomBarView.java
package com.raival.fileexplorer.common.view;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.raival.fileexplorer.R;
import com.raival.fileexplorer.util.PrefsUtils;

public class BottomBarView extends LinearLayout {
    public BottomBarView(Context context) {
        super(context);
    }

    public BottomBarView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public BottomBarView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void addItem(String tag, int icon, OnClickListener clickListener) {
        View view = ((LayoutInflater) getContext()
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE))
                .inflate(R.layout.bottom_bar_menu_item, this, false);
        view.setOnClickListener(clickListener);
        view.setTooltipText(tag);
        TextView label = view.findViewById(R.id.label);
        if (PrefsUtils.Settings.getShowBottomToolbarLabels()) {
            label.setText(tag);
        } else {
            label.setVisibility(GONE);
        }
        ImageView image = view.findViewById(R.id.icon);
        image.setImageResource(icon);
        addView(view);
    }

    public void addItem(String tag, View view) {
        view.setTooltipText(tag);
        addView(view);
    }

    public void clear() {
        removeAllViews();
    }

    public void onUpdatePrefs() {
        for (int i = 0; i < getChildCount(); i++) {
            final View view = getChildAt(i);
            final View label = view.findViewById(R.id.label);
            if (label != null) {
                label.setVisibility(PrefsUtils.Settings.getShowBottomToolbarLabels() ? VISIBLE : GONE);
            }
        }
    }
}

File: app/src/main/java/com/raival/fileexplorer/common/view/BottomBarView.kt
package com.raival.fileexplorer.common.view

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import com.raival.fileexplorer.R
import com.raival.fileexplorer.util.PrefsUtils

class BottomBarView : LinearLayout {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    fun addItem(tag: String?, icon: Int, clickListener: OnClickListener?) {
        val view = (context
            .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater)
            .inflate(R.layout.bottom_bar_menu_item, this, false)
        view.setOnClickListener(clickListener)
        view.tooltipText = tag
        val label = view.findViewById<TextView>(R.id.label)
        if (PrefsUtils.Settings.showBottomToolbarLabels) {
            label.text = tag
        } else {
            label.visibility = GONE
        }
        val image = view.findViewById<ImageView>(R.id.icon)
        image.setImageResource(icon)
        addView(view)
    }

    fun addItem(tag: String?, view: View) {
        view.tooltipText = tag
        addView(view)
    }

    fun clear() {
        removeAllViews()
    }

    fun onUpdatePrefs() {
        for (i in 0 until childCount) {
            val view = getChildAt(i)
            val label = view.findViewById<View>(R.id.label)
            if (label != null) {
                label.visibility =
                    if (PrefsUtils.Settings.showBottomToolbarLabels) VISIBLE else GONE
            }
        }
    }
}